### PR TITLE
python: Return ProgramError

### PR DESCRIPTION
**Pull request by tebeka**
Return user code errors as ProgramError with call stack.

Fixes ENG-1129

### CODE REVIEW

**Comment by ig-ra** [comment]
@tebeka , could u please add tests for this?

**Review by ig-ra — CHANGES_REQUESTED** [review]
@tebeka , could u please add tests for this?

**Comment by tebeka** [comment]
> @tebeka , could u please add tests for this?

Correct, will do.

**Comment by tebeka** [comment]
> > @tebeka , could u please add tests for this?
> 
> Correct, will do.

Done

File: runtimes/pythonrt/pythonrt_test.go [hunk 1]
```diff
@@ -253,3 +264,51 @@ func TestCleanup(t *testing.T) {
...
+	xid := sdktypes.NewExecutorID(runID)
+	fn, err := sdktypes.NewFunctionValue(xid, "handle", nil, nil, mod)
+	require.NoError(t, err, "new function")
+
+	kwargs := map[string]sdktypes.Value{}
+
+	_, err = svc.Call(ctx, fn, nil, kwargs)
+	require.Error(t, err)
+	// There no way to check that err is a ProgramError since it's wrapped by unexported programError
+	require.Contains(t, err.Error(), "division by zero")
```
Comment (ig-ra) [inline runtimes/pythonrt/pythonrt_test.go:314]: could we check that it contains traceback/function/line?

Comment (tebeka) [reply runtimes/pythonrt/pythonrt_test.go:314]: Hmm, the stack trace is:
```
division by zero
         [0] <frozen runpy>:198.1 _run_module_as_main
         [1] <frozen runpy>:88.1 _run_code
         [2] /tmp/ak-2638291852/ak_runner/__main__.py:204.1 <module>
         [3] /tmp/ak-2638291852/ak_runner/__main__.py:135.1 run
```

I guess we want the user code there as well, checking...

Comment (tebeka) [reply runtimes/pythonrt/pythonrt_test.go:314]: Thanks @ig-ra, now the traceback is fill.

**Review by ig-ra — APPROVED** [review]

**Pull request merged**